version-detection/version-detection.md

```diff
@@ -11,87 +11,61 @@ to parse the data.  The GEDCOM format itself does include a version identifier i
 a parser can detect the correct version and parse the file accordingly.  This document specifies a version
 detection algorithm and provides references to the more specific format specifications.
 
-## Character Width and Byte Order Detection
+## Commonalities
 
-To efficiently locate the version in the content, one can first determine the character width and byte order
-used in the file.  (The version detection algorithm can be done without this by trying all three possibilities
-separately.)
+Every GEDCOM file, regardless of version, is a text file that conforms to the following general structure:
 
-A GEDCOM file starts with the character "0" in some character set, possibly prefixed by a
-Byte Order Mark (BOM) indicating the character set.  The actual character set, however, does not matter
-as version detection only requires knowing the character width and byte order.
+- The document is organized as a sequence of **lines** separated by ASCII line terminators (U+000A, U+000D, or both). Blank lines are ignored.
+- Each line contains several components, separated by spaces (U+0020):
+  1. A required **level**, which is a non-negative integer encoded using ASCII digits.
+    - The first line's level is 0.
+    - Each other line's level is between 0 and 1 + *x* inclusive, where *x* is the level of the immediately preceding line.
+  2. An optional **identifier**, which begins and ends with a COMMERCIAL AT (U+0040 `@`) and contains no internal line terminators or COMMERCIAL ATs.
+  3. A required **tag**, which is a string of one or more printable ASCII characters (U+0021 through U+007E), excluding the COMMERCIAL AT (U+0040 `@`).
+  4. An optional **value**, which is a string of zero or more characters; all characters are permitted except ASCII line terminators (U+000A and U+000D).
+- Each line represents a structure. Any line with level *x* > 0 represents a substructure of the structure represented by the nearest preceding line with level *x* − 1.
 
-Thus, the character width and byte order can be determined by reading the first two characters of the file,
-and using the table below:
+Beyond these commonalities, each GEDCOM version differs in details and is described by its own specification.
+Version specifications typically describe both the meaning of various structures
+and provide additional serialization details, including whether blank lines are allowed, whether multiple spaces can be used to separate line components, limits on tag, identifier, and value lengths and characters, etc.
 
-Initial bytes (hex) | Width | Order
-------------------- | ----- | -------------
-FF FE               | 2     | LE (Little-endian)
-30 00               | 2     | LE (Little-endian)
-FE FF               | 2     | BE (Big-endian)
-00 30               | 2     | BE (Big-endian)
-Otherwise           | 1     | (N/A)
 
-## Version Detection
+## Version detection
```

Comment:
I have no problem with the _addition_ of the new text above, as long as it is informative not normative.
However, the text removal above and changes below are, in my view, problematic as it lost all discussion of character set.  One cannot do string matching without a discussion of character set.  As such I'd prefer to keep byte matching, even if informative text such as above is added as an introduction.